src/examples/pongxr/scene.js

```diff
@@ -0,0 +1,200 @@
+/*****
+WHEN CORE IS WORKING:
+- TEST IN THE SOCIAL PLACES
+- RESPONSIVE DESIGN: PHONE
+- RESPONSIVE DESIGN: MAGIC LEAP
+*/
+
+import { Scene, Object3D } from "three";
+import XRInput from "../../engine/xrinput"
+import PeerConnection from '../../engine/networking/PeerConnection'
+import State from "../../engine/state";
+import Physics from "../../engine/physics/physics"
+
+import HostBot from "../../engine/util/hostbot"
+import Level from "./level"
+import Paddle from "./paddle"
+import Ball from "./ball"
+import PlacementCube from "./placementcube"
+
+const scene = new Scene();
+const networking = new PeerConnection(scene);
+const hostBot = new HostBot(networking);
+let ball, placementCube;
+let paddles = [];
+
+// custom States & events
+State.eventHandler.registerEvent('gameover');
+const GameState = { placement: 1, play: 2 }
+
+Physics.enableDebugger(scene);
+
+const createPongLevel = (placementCube) =>
+{
+    const targetPosition = placementCube.position;
+    const targetRotation = placementCube.rotation;
+
+    State.GameState = GameState.play;
+    const curPosRot = { position: targetPosition, rotation: targetRotation }
+    const level = new Level(curPosRot);
+    scene.add(level);
+    networking.remoteSync.addLocalObject(level, { type: "level", posRot: curPosRot }, true);
+
+    // dumb hack for the other side 
+    placementCube.scale.set(0, 0, 0);
+
+    // local
+    scene.remove(placementCube);
+    // remote placement cube removal, not currently working on metachromium
+    networking.remoteSync.removeSharedObject(3);
+
+    // BALL
+    if (State.isMaster)
+    {
+        ball = new Ball(targetPosition, true);
+        ball.initPos = targetPosition;
+        scene.add(ball);
+        networking.remoteSync.addLocalObject(ball, { type: "ball", position: targetPosition }, true);
+    }
+}
+
+const initPlacement = () =>
+{
+    State.GameState = GameState.placement;
+    placementCube = PlacementCube();
+    scene.add(placementCube);
+    networking.remoteSync.addSharedObject(placementCube, 3, true);
+}
+
+/// GAME STATE
+
+State.eventHandler.addEventListener("gameover", (e) =>
+{
+    if (ball != undefined)
+    {
+        ball.reset();
+    } else
+    {
+        console.error("can't reset; no ball found!");
+    }
+});
+
+/// INPUT
+
+let doubleClick = false;
+
+State.eventHandler.addEventListener("select", e =>
```

Comment:
If you had separate callbacks for placement and play, could you bind (and unbind) the appropriate listeners in `createPongLevel` and `initPlacement`? That might remove the need for the GameState enum.

Reply:
I wonder if the price would be spaghettification of input logic - I kinda like that all input is defined in one place, readability-wise if for no other reason.

Reply:
You're right. Once I got to this section of the code I was like, "Oh, now it all makes sense!" This handler brings everything together. Breaking it up would probably make things difficult to read/understand.